### DISCUSSION TITLE
GrapesJS out in the wild

### DISCUSSION THREAD

**Discussion by chrisedington** [GrapesJS/grapesjs#5255] [Show and tell]
Hi,

I wanted to check out some examples of how people are using GrapesJS and how they may have extended it or customized it to their needs.

Would be cool if anyone with a public facing (I guess even a screenshot) site commented on this issue to build a list of current uses of this great extension!

**Comment by arthuralmeidap** [comment]
Unfortunately, I can disclosure the app built on top of GrapesJS, but I have been working on an UI tool to build E-mail marketing. We have a lot of customizations/integrations to improve the existing greatness of GrapesJS like creating a custom Asset Manager; Social blocks (to add social medias to the email);

**Comment by tristanMatthias** [comment]
I'm using in [Origami CMS](https://github.com/origami-cms/cms)!

**Comment by artf** [comment]
@tristanMatthias wow! seems quite promising 👏

**Comment by tristanMatthias** [comment]
Thanks @artf! I need a lot of help 😅

**Comment by noogen** [comment]
Still a WIP, I'm using it to build new form components: https://niiknow.github.io/grapesjs-components-farmer/

Mainly focus on getting unique Name field, which are important for Form submission.  Also implement doTjs for templating, so I can work on supporting additional CSS framework with initial support of Bootstrap4.

**Comment by artf** [comment]
@noogen great, thanks for sharing

**Comment by loganvolkers** [comment]
We use GrapesJS to power our CMS at http://saasquatch.com

*Widgets*

 - GrapesJS powers the editing of the content in our in-app javascript widget
 - We have customized most of the toolbars, removed most of the default components, and have it made up of only custom web components
 - Plugged in https://cloudinary.com/ for asset management
 - It works pretty great with Web Components and https://stenciljs.com/ ❤️ 
 - Recommendation: Use GrapesJS with Web Components

![2019-03-06 18_22_40-](https://user-images.githubusercontent.com/1157086/53927888-0e620b80-403d-11e9-9fe8-f0cbc1366b13.png)


*Emails*

 - We also use it for HTML email editing, but find that it isn't great (because HTML emails are a nightmare to get right, not directly a fault of GrapesJS....) We're looking to switch to GrapesJS + MJML or https://beefree.io/ to make the end-user experience more bearable.
 - Recommendation: Avoid GrapesJS for plain HTML email templates

![image](https://user-images.githubusercontent.com/1157086/53928211-21291000-403e-11e9-8458-8ec81e22ff02.png)

**Reply by pedroclayman** [reply]
Hey @loganvolkers, I know it's been years since your comment, but I was wondering - which solution did you go for for the email editor use case? Was it beefree.io, grapesjs+MJML or something else? How did it work out? Thank you

**Reply by loganvolkers** [reply]
> Hey @loganvolkers, I know it's been years since your comment, but I was wondering - which solution did you go for for the email editor use case? Was it beefree.io, grapesjs+MJML or something else? How did it work out? Thank you

We're still on GrapesJS for emails.

I wish that MJML had more momentum. Unfortunately I don't even have a recommendation for which email builder to use:

- Builders that can import HTML and use the HTML as the persisted source of truth are walkways flakey and struggle with markup (e.g. GrapesJS)
- Builders that provide higher level components as the persisted source of truth (e.g. BeeFree) lock you into their component format for all time. 

On the web there is the web components standard in all browsers to help you escape the lock in. We wrote saasquatch/raisins and migrated seamlessly from Grapes because of that standard technology for components.

There is no standard system for email components, despite some hope that MJML could become a defacto standard.

So, when you're picking an email editor, you're picking flakiness in editing (e.g. Grapes) or infinite vendor lock in (e.g. BeeFree).

Good luck with the choice and let me know if you can escape that trade-off

**Reply by Jogai** [reply]
Just discovered [EmailBuilderJs](https://github.com/usewaypoint/email-builder-js). No mjml stuff tough...

**Comment by artf** [comment]
Amazing work @loganvolkers thanks for pointing out your use cases

**Comment by gabrigcl** [comment]
I'm using grapesjs as the core of our online ebook creator SaiteBooker: [www.saitebooker.com.br.](http://www.saitebooker.com.br) It's still in development and this public version is a prototype.

**Comment by artf** [comment]
@gabrigcl fantastic use case 👏

**Comment by trafalmejo** [comment]
Hi Guys,

I am using Grapesjs to help people to create GUIs to control interactive installations:
[OSCAR create responsive and beauty GUIs to control your installations](https://www.youtube.com/watch?v=ZBBGbX43hbY)

**Comment by lcsdms** [comment]
> We use GrapesJS to power our CMS at http://saasquatch.com
> 
> _Widgets_
> 
> * GrapesJS powers the editing of the content in our in-app javascript widget
> * We have customized most of the toolbars, removed most of the default components, and have it made up of only custom web components
> * Plugged in https://cloudinary.com/ for asset management
> * It works pretty great with Web Components and https://stenciljs.com/ ❤️
> * Recommendation: Use GrapesJS with Web Components
> 
> ![2019-03-06 18_22_40-](https://user-images.githubusercontent.com/1157086/53927888-0e620b80-403d-11e9-9fe8-f0cbc1366b13.png)
> 
> _Emails_
> 
> * We also use it for HTML email editing, but find that it isn't great (because HTML emails are a nightmare to get right, not directly a fault of GrapesJS....) We're looking to switch to GrapesJS + MJML or https://beefree.io/ to make the end-user experience more bearable.
> * Recommendation: Avoid GrapesJS for plain HTML email templates
> 
> ![image](https://user-images.githubusercontent.com/1157086/53928211-21291000-403e-11e9-8458-8ec81e22ff02.png)

Looks like there's a new one for MJML available: https://grapesjs.com/demo-mjml.html (https://github.com/artf/grapesjs-mjml)

**Comment by edvinasba1** [comment]
@froderf May we see some screenshots or maybe a demo?

**Comment by robhoward** [comment]
We're upgrading to GrapesJS for our Landing Page and Email designers in [DailyStory](https://www.dailystory.com).

We made quite a few changes, mostly to turn-off or disable certain functionality while also writing all new components from the ground-up so we could control the editing experience (we limit what you can/cannot do in the designer), e.g. where blocks and components can be placed. 

And, all the landing page components are built around Bootstrap. Just keeps things super-simple for our customers.

![image](https://user-images.githubusercontent.com/946840/59806731-0a3fbe00-92bb-11e9-852b-21f7fee7604e.png)

We then had to start from scratch and write another set of components / blocks from the ground up for email templating with tables to work in an email-friendly way while controlling what could/could not be done in the editor. We needed different types of components for different situations, e.g. a non-editable/removable layout container table vs. a draggable content section table.

![image](https://user-images.githubusercontent.com/946840/59807214-b504ac00-92bc-11e9-9ec7-4c4c7c0cc277.png)

We also integrated our own file manager.

![image](https://user-images.githubusercontent.com/946840/59806768-280d2300-92bb-11e9-88d5-e7340d813db4.png)

We're launching our landing page editor to our customers in 2 weeks (still putting the finishing touches on it). We're planning to launch our updates to the email editor in late July. We started both these projects about 9 months ago (worked on them on-and-off). I'll share some screen shots of the email editor later.

Great project @artf

**Comment by artf** [comment]
Amazing work @robhoward thanks for sharing

**Comment by DavidSporer** [comment]
We're using GrapesJs as our email editor for [Passcreator](https://www.passcreator.com).
Most changes were simply styling and a couple of changes regarding to the column blocks of emails.
We still need to revamp it since we didn't spend too much time integrating it last year but it's working quite well.
![image](https://user-images.githubusercontent.com/1636571/62518297-89686180-b829-11e9-83ef-ac1ac275c786.png)

**Comment by tasham16** [comment]
> We use GrapesJS to power our CMS at http://saasquatch.com
> 
> _Widgets_
> 
> * GrapesJS powers the editing of the content in our in-app javascript widget
> * We have customized most of the toolbars, removed most of the default components, and have it made up of only custom web components
> * Plugged in https://cloudinary.com/ for asset management
> * It works pretty great with Web Components and https://stenciljs.com/ ❤️
> * Recommendation: Use GrapesJS with Web Components
> 
> ![2019-03-06 18_22_40-](https://user-images.githubusercontent.com/1157086/53927888-0e620b80-403d-11e9-9fe8-f0cbc1366b13.png)
> 
> _Emails_
> 
> * We also use it for HTML email editing, but find that it isn't great (because HTML emails are a nightmare to get right, not directly a fault of GrapesJS....) We're looking to switch to GrapesJS + MJML or https://beefree.io/ to make the end-user experience more bearable.
> * Recommendation: Avoid GrapesJS for plain HTML email templates
> 
> ![image](https://user-images.githubusercontent.com/1157086/53928211-21291000-403e-11e9-8458-8ec81e22ff02.png)

@loganvolkers do you have a coded example of how you integrated the web components with grapesjs?

**Comment by ThatoLesetla** [comment]
Yes I do, I’ll have to upload it on GitHub then I’ll send
It to you. I was working on a website builder so it appeared to be a great
framework to use

**Comment by ThatoLesetla** [comment]
I’m was also struggling with saving the content on the database. Is it
possible to chat on WhatsApp?

**Comment by tasham16** [comment]
> Yes I do, I’ll have to upload it on GitHub then I’ll send It to you. I was working on a website builder so it appeared to be a great framework to use
> […](#)
> On Thu, 03 Oct 2019 at 22:23, tasham16 ***@***.***> wrote: We use GrapesJS to power our CMS at http://saasquatch.com *Widgets* - GrapesJS powers the editing of the content in our in-app javascript widget - We have customized most of the toolbars, removed most of the default components, and have it made up of only custom web components - Plugged in https://cloudinary.com/ for asset management - It works pretty great with Web Components and https://stenciljs.com/ ❤️ - Recommendation: Use GrapesJS with Web Components [image: 2019-03-06 18_22_40-] <https://user-images.githubusercontent.com/1157086/53927888-0e620b80-403d-11e9-9fe8-f0cbc1366b13.png> *Emails* - We also use it for HTML email editing, but find that it isn't great (because HTML emails are a nightmare to get right, not directly a fault of GrapesJS....) We're looking to switch to GrapesJS + MJML or https://beefree.io/ to make the end-user experience more bearable. - Recommendation: Avoid GrapesJS for plain HTML email templates [image: image] <https://user-images.githubusercontent.com/1157086/53928211-21291000-403e-11e9-8458-8ec81e22ff02.png> @loganvolkers <https://github.com/loganvolkers> do you have a coded example of how you integrated the web components with grapesjs? — You are receiving this because you commented. Reply to this email directly, view it on GitHub <#1798?email_source=notifications&email_token=AK2F6JKLLSQTQMOBCUQZGUDQMZIF3A5CNFSM4GXRC75KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEAJPHVA#issuecomment-538112980>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AK2F6JM5F4NG6P7U37IU2N3QMZIF3ANCNFSM4GXRC75A> .

@ThatoLesetla I am doing something similar! Yes please let me know if you upload the code.

**Comment by rlopzc** [comment]
@loganvolkers man that looks amazing! 

Do you have an example or the code uploaded about grapejs + web-components?

**Comment by tasham16** [comment]
@ThatoLesetla any updates?

**Comment by pvbergen** [comment]
I can add our use case to this issue: https://www.drupal.org/project/pagedesigner

We have created a Drupal 8 module based on grapesjs for visual editing of pages. We utilize grapesjs as the frontend and wrote a custom interface to Drupals backend. This allows us to layout individual nodes with the WYSIWYG interface provided by grapesjs and store the components and traits in a structured tree.

Here's a screenshot of one our demo pages:
![image](https://user-images.githubusercontent.com/3135880/69811147-bb913b00-11ed-11ea-91f0-608a5c748bb2.png)
We customized the css quite a bit, but the js of grapes is untouched. 

I had a talk end of October at DrupalCon Amsterdam [(Making Drupal as easy to us as Squarespace)](https://www.youtube.com/watch?v=na-ZTwMj2iE) and people were very impressed. Lot's of that is thanks to the great work @artf and all the other contributors provide for grapesjs. Thanks very much!

**Comment by artf** [comment]
Thanks @pvbergen for sharing the use case and making your module available for others. Great talk and the integration seems quite solid 👏

**Comment by Jogai** [comment]
We're sending more than 1.5 million emails each week to more than 300K subscribers in 20 lists of subscribers (its not spam, they subscribed, maybe after a bit encouragement from us). Our system uses a template to generate a prefilled newsletter which is rendered in grapesjs. Users drag stuff around and then send it.

**Comment by GoodPHP** [comment]
We're creating websites builder projects. We often use GrapesJS like editor for pages or like editor for newsletter.

We provide services on GrapesJS: https://devfuture.pro/grapesjs-development/

Some examples:

**GrapesJS new design**
GIT: https://github.com/GoodPHP/GrapesJS-devfuture
Demo: https://devfuture.pro/live/new-design-grapesjs/

<img src="https://devfuture.pro/wp-content/uploads/2020/03/new-editor-devfuture.gif">

**GrapesJS Multiple Pages**

<img src="https://devfuture.pro/wp-content/uploads/2020/03/grapesjs-multiple-pages.gif">

**Comment by HansSchouten** [comment]
Thanks for the awesome work on creating GrapesJS!

In my web projects I love to have the power of PHP frameworks like Laravel to create complex custom dynamic components, but want to offer my clients the ease of managing pages using a pagebuilder. That is why I am developing [PHPageBuilder](https://github.com/HansSchouten/PHPageBuilder), which allows to add GrapesJS to any PHP project. With [Laravel Pagebuilder](https://github.com/HansSchouten/Laravel-Pagebuilder) specifically for using GrapesJS in any Laravel project.

![PHPageBuilder](https://user-images.githubusercontent.com/5946444/80287557-b1d71980-8732-11ea-945c-a9d9e1792c1a.png)

**Comment by artf** [comment]
@HansSchouten truly amazing, and thanks for the open-source libraries

**Comment by amodliszewski** [comment]
At [Simple LMS](https://simple-lms.com) we have decided to use GrapesJS as page editor for our clients. We have also created additional integration with our built-in file library, allowing to pick images that are available on website. We also did changed layout a bit.

<img width="1920" alt="Zrzut ekranu 2020-05-6 o 12 52 13" src="https://user-images.githubusercontent.com/1456990/81169264-dd5ecd00-8f98-11ea-97d4-e5a977216bc7.png">

But there is a lot to do with it, we are still missing some easy to use components and integration with our video library for video component.

<img width="1919" alt="Zrzut ekranu 2020-05-6 o 12 51 57" src="https://user-images.githubusercontent.com/1456990/81169257-d932af80-8f98-11ea-96ff-ad246eb84ce5.png">

**Comment by BerkeAras** [comment]
Using it in experiencedCMS. An intelligent Content Management System with an Artificial Intelligence.

**Comment by ChrisB9** [comment]
I am currently developing a Page-Builder for the CMS/PIM pimcore.org inspired by @HansSchouten 
https://github.com/ChrisB9/PimcorePageBuilderBundle
It is still in early development but already working

![image](https://user-images.githubusercontent.com/7099583/90976440-65171a00-e53d-11ea-9a66-89bc71615e46.png)

**Comment by fzerman** [comment]
I am developing a page builder for my web builder project. 
![Screenshot from 2020-12-31 13-43-37](https://user-images.githubusercontent.com/59289764/103407133-74067c80-4b6e-11eb-848c-f8f0085c72ed.png)

I will add my custom managers and a package index repo like a hub. Thanks for this framework.

**Comment by millord** [comment]
Hello guys!!

How are you doing the integration with the cms. In my case I would do it with ghost or sanity. Where I can find how to integrate it? In the Storage Manager section?

**Comment by ibtissammoufid** [comment]
Hello everyone, we want to build a page builder using Garpejs and angular, has anyone ever used it with angular? I'm a beginner and I would appreciate some guidelines or recommendations, thanks!

**Comment by BerkeAras** [comment]
@ibtissammoufid As a beginner, I would not start using angular with GrapesJS. Angular is very complex, GrapesJS can be very complex – not very beginner friendly!

**Comment by BerkeAras** [comment]
But there is a Medium Blog-Post about integrating GrapesJS in Angular, but unfortunately not in english (in spanish). Try to translate it: https://medium.com/@camiloht23/integración-de-grapesjs-con-angular-11ebf3bb80c5

**Comment by ibtissammoufid** [comment]
@BerkeAras unfortunately, I have to work with Angular to build a page builder, and since it would be difficult to start from scratch, I thought GrapesJs would be a good starting point, do you know any better options, thanks a lot for your answer!

**Comment by Jogai** [comment]
It may not be a beginner task, but you should be able to integrate it into an angular environment. That said, this is the wrong issue to have this discussion.

**Comment by anlumo** [comment]
My project has finally been released, so I can show it off now:
![image](https://user-images.githubusercontent.com/644763/127492563-ba8fd924-0904-4088-ba1f-993198aa2046.png)
![image](https://user-images.githubusercontent.com/644763/127493056-abc0a309-8ad5-4f43-ba73-95b9ae9a2db0.png)

It's part of our tabletop RPG battlemap editor [DungeonFog](https://www.dungeonfog.com) and used for creating notes and adventure books for game masters. The idea is that you can print them to PDF from the browser, so it's kinda like a simpler InDesign.

**Comment by artf** [comment]
Awsome project @anlumo and thanks also for being an active contributor on GrapesJS, I hope to continue seeing you here :)

**Comment by annus97** [comment]
> Hello everyone, we want to build a page builder using Garpejs and angular, has anyone ever used it with angular? I'm a beginner and I would appreciate some guidelines or recommendations, thanks!

Check this article @ibtissammoufid: https://esketchers.com/integration-of-grapesjs-with-angular/

**Comment by Mambatukaa** [comment]
@anlumo Hello Sir 👋 . I'm trying to make a page structure on grapesjs. Could you share with me how to add pagination in DungeonFog's GM-notes?

**Comment by anlumo** [comment]
> @anlumo Hello Sir 👋 . I'm trying to make a page structure on grapesjs. Could you share with me how to add pagination in DungeonFog's GM-notes?

Unfortunately, CSS doesn't allow for controlled pagination (it's not designed for print), so this is all a big hack.

Every page is a div sized according to the paper size (directly in millimeters) and the wrapper only allows page elements in it. The CSS disables all print margins and replaces them with regular CSS padding.

It's not possible to overflow text from one element to another in CSS, so page wrapping has to be done manually by the user.

Additionally, there's a bug in Chrome concerning page margins, which I reported [here](https://bugs.chromium.org/p/chromium/issues/detail?id=1208103). It's still open after about 1.5 years.

**Comment by Jogai** [comment]
Guys, this belongs in an issue on its own, not here at least

**Comment by ronaldohoch** [comment]
I am very happy to be able to show you our version of GrapesJs, I have been working to adapt GrapesJs with my colleague @gustavohleal for over a year and it is already being used by approximately 100 customers. Let's release it little by little to the others ^^

We have two versions of grapesJs, Landing Page version and Email Marketing version using mjml, In addition to a css work to customize the editor, we work developing plugins, traits and custom styles as well, in addition to reporting many bugs and opening several discussions as well. and of course, helping the community whenever possible!

Thank you, @artf  and the community ♥

![image](https://user-images.githubusercontent.com/2287371/197536233-c21d30a1-38ae-4be5-942c-51860cda4aa2.png)

![image](https://user-images.githubusercontent.com/2287371/197536932-d2b9260f-ed78-4aef-ad0e-2b9f71cf728d.png)

![image](https://user-images.githubusercontent.com/2287371/197537038-ca68468c-f81f-4d25-ad50-3bcfc4e02075.png)

**Comment by GoodPHP** [comment]
Discover newly preset for GrapesJS. White colors with full functionality for webpage / newslatter.

**Whitener**

![Whitener](https://gjs.market/_next/image?url=https%3A%2F%2Fapi.gjs.market%2Fstorage%2F116%2FWhitener-Preset-GrapesJS.png&w=3840&q=100)

Download / Demo:
https://gjs.market/products/whitener-preset-grapesjs

**Comment by Mambatukaa** [comment]
> > @anlumo Hello Sir 👋 . I'm trying to make a page structure on grapesjs. Could you share with me how to add pagination in DungeonFog's GM-notes?
> 
> Unfortunately, CSS doesn't allow for controlled pagination (it's not designed for print), so this is all a big hack.
> 
> Every page is a div sized according to the paper size (directly in millimeters) and the wrapper only allows page elements in it. The CSS disables all print margins and replaces them with regular CSS padding.
> 
> It's not possible to overflow text from one element to another in CSS, so page wrapping has to be done manually by the user.
> 
> Additionally, there's a bug in Chrome concerning page margins, which I reported [here](https://bugs.chromium.org/p/chromium/issues/detail?id=1208103). It's still open after about 1.5 years.

Could you share with me that GM-notes source code. I want to customize grapesjs like yours. But I don't know how to customize like that.

**Comment by GoodPHP** [comment]
Marketplace for plugins / presets on GrapesJS

Website:
https://gjs.market

There are any users able to post feature for community.

**Comment by aclarembeau** [comment]
We're using it too on our website: https://www.touchlead.app/ 
We use it to allow users building custom branded emails in our marketing automation tool
![image](https://github.com/user-attachments/assets/85937497-4def-40d8-91fd-06467b58d3c8)

**Comment by toiyabe** [comment]
I'm using GrapesJS as our page and template builder with our open source CMS called [Cosmos](https://cosmos.moonrise.net/) (see also the [Git repo](https://github.com/MoonriseSoftwareCalifornia/CosmosCMS)).  The product uses GrapesJS, CKEditor 5, VS Code (Monaco), Filerobot image editor, and Filepond together.

I have to say GrapesJS is really a wonderful thing.  Works really well and the documentation is super helpful.

Our rendition of GrapesJS automatically supports Bootstrap 4 and 5, and Tailwind--more UI frameworks to follow. The way it works is that it detects what UI is being used and automatically loads the plugin(s) for it.

I'm also starting to add my own custom components as well.  I have a video and screenshot below.  

Demonstration video:  https://www.youtube.com/watch?v=mVGPlbnbC5c

![image](https://github.com/user-attachments/assets/0b274c01-f5ba-48ea-9be6-b8dd9183ad6e)

**Reply by artf** [reply]
well done @toiyabe 👏

**Comment by GJSDocs** [comment]
Hi everyone 👋

Created GJSDocs — a lightweight document builder layer built on top of GrapesJS that focuses specifically on:

template-based document editing
dynamic variables (e.g. {{name}}, {{price}})
SaaS integration use cases (invoices, contracts, emails)
faster setup of document systems inside products
integration with crm (Airtable, HubSpot, Google Sheets, Salesforce, Stripe, Zapier, and Rest API)

Live demo: https://gjsdocs.com/

<img width="1440" height="719" alt="Знімок екрана 2026-04-21 о 20 58 03" src="https://github.com/user-attachments/assets/5ec9df3a-61ac-4bf4-b62d-f9480dfeb5df" />
